File: src/main/java/kr/entree/spigradlesample/SpigradleSample.java
package kr.entree.samplespigot;

import kr.entree.spigradle.PluginMain;
import org.bukkit.plugin.PluginDescriptionFile;
import org.bukkit.plugin.java.JavaPlugin;
import org.bukkit.plugin.java.JavaPluginLoader;

import java.io.File;

/**
 * Created by JunHyung Lim on 2020-06-08
 */
@PluginMain
public class SpigradleSample extends JavaPlugin {
    public SpigradleSample() {
    }

    public SpigradleSample(JavaPluginLoader loader, PluginDescriptionFile description, File dataFolder, File file) {
        super(loader, description, dataFolder, file);
    }

    @Override
    public void onEnable() {
        System.out.println("Hello!");
    }
}

File: spigot-kotlin/src/main/kotlin/kr/entree/samplespigot/SpigradleSample.kt
package kr.entree.samplespigot

import kr.entree.spigradle.PluginMain
import org.bukkit.plugin.PluginDescriptionFile
import org.bukkit.plugin.java.JavaPlugin
import org.bukkit.plugin.java.JavaPluginLoader
import java.io.File

@PluginMain
class SpigradleSample : JavaPlugin {
    constructor() : super()

    // Constructor for MockBukkit. if you don't need it, use: "class SpigradleSample() : JavaPlugin()" without both constructors.
    constructor(loader: JavaPluginLoader, description: PluginDescriptionFile, dataFolder: File, file: File?) : super(loader, description, dataFolder, file
            ?: File(""))

    override fun onEnable() {
        logger.info("This is Spigot plugin!");
    }
}